src/paginator-spec.js

```diff
@@ -0,0 +1,486 @@
+        });
+    });
+
+    describe('fetchPage', function() {
+      describe('(LimitOffsetPagination)', function() {
+        beforeEach(function() {
+          const options = {
+            limitQueryParam: 'l',
+            offsetQueryParam: 'o',
+            style: Paginator.styles.limitOffset
+          };
+          paginator = new Paginator(request.spy, null, {}, options);
+        });
+
+        it('set the limit query parameter using the configured property name',
+          function() {
+            const queryParams = paginator.fetchPage(1)
+              .then(request.getQueryParams);
+
+            return expect(queryParams).to.eventually.have.property('l');
+          });
+
+        it('set the offset query parameter using the configured property name',
+          function() {
+            const queryParams = paginator.fetchPage(1)
+              .then(request.getQueryParams);
+
+            return expect(queryParams).to.eventually.have.property('o');
+          });
+      });
+    });
+
+    describe('fetchPage', function() {
+      describe('(LimitOffsetPagination)', function() {
+        const style = Paginator.styles.limitOffset;
+        const queryParams = {
+          offset: 10,
+          limit: 5
+        };
+
+        beforeEach(function() {
+          paginator = new Paginator(request.spy, null, queryParams, { style });
+        });
+
+        it('provides a limit query parameter', function() {
+          const queryParams = paginator.fetchPage(1)
+            .then(request.getQueryParams);
+
+          return expect(queryParams).to.eventually.have.property('limit');
+        });
+
+        it('provides an offset query parameter', function() {
+          const queryParams = paginator.fetchPage(1)
+            .then(request.getQueryParams);
+
+          return expect(queryParams).to.eventually.have.property('offset');
+        });
+
+        it('doesn\'t provide a page query parameter', function() {
+          const queryParams = paginator.fetchPage(1)
+            .then(request.getQueryParams);
+
+          return expect(queryParams).to.eventually.not.have.property('page');
+        });
+
+        it('calulates the offset based on the given page', function() {
+          const offset = paginator.fetchPage(4)
+            .then(request.getOffset);
+
+          return expect(offset).to.eventually.equal(15);
+        });
+      });
+    });
+
+    describe('fetchPageCount', function() {
+      beforeEach(function() {
+        paginator = new Paginator(request.spy);
+      });
+
+      it('returns a promise that successfully resolves', function() {
+        const promise = paginator.fetchPageCount();
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('returns a promise with the paginator\'s page count', function() {
+        const pageCount = paginator.fetchPageCount();
+
+        return expect(pageCount).to.eventually.equal(10);
+      });
+    });
+
+    describe('clone', function() {
```

Comment:
It's not needed. We decided that the paginator wouldn't have a method to reset cache, but instead we'd just create a new paginator instance. This was a quick helper to facilitate that.